### DISCUSSION TITLE
How to access another element's x-data object?

### DISCUSSION THREAD

**Discussion by OvalMedia** [alpinejs/alpine#4166] [1. Help]
AlpineJS noob here: I have an x-data object on a div element. In an adjacent button I want to access that data in an onclick event.
Roughly: 
```
<div x-data={x:1}><div>
<button @click="submitHandler">Submit</button>
```
How can I access the x-data object from within the submitHandler function?

Thank you

**Comment by ekwoka** [comment]
For what purpose?

Can the elements not be in the same tree with the data?

There are ways, but this is more likely to be an indication of poor component design.

**Comment by OvalMedia** [comment]
The goal is to gather form data and send it via ajax. No, they can't be in the same tree.

**Reply by ekwoka** [reply]
Are they not already in the same form?

Can you not listen for submit on the form and get the formdata from the form itself?

why do you need to access the internal state?

**Reply by SimoTod** [reply]
you don't need alpine for the button, this is already supported by default in your browser by using the form attribute on the button tag: https://codepen.io/SimoTod/pen/bGJOOpo?editors=1111
